src/components/Stepper/sgds-stepper.ts

```diff
@@ -2,59 +2,98 @@ import { html } from "lit";
+ * @event sgds-last-step - Emitted right before the last step is reached. Event is fired when lastStep method is called.
+ * @event sgds-first-step - Emitted on hide after animation has completed. Event is fired when firstStep method is called.
+ * @event sgds-arrived - Emitted right after the activeStep has updated its state, when upcoming step has arrived and its slot are rendered.
+ * @event sgds-reset - Emitted right before the step is reset to its defaultActiveStep. Event is fired when reset method is called.
+ *
+ */
 @customElement("sgds-stepper")
 export class SgdsStepper extends SgdsElement {
-  static styles = [SgdsElement.styles, styles];
+  static styles = [SgdsElement.styles];
 
+  /** The header name for steps in chronological order
+   */
   @property({ type: Array })
-  steps = [
-    {
-      title: 1,
-      stepHeader: "Marker title 1"
-    },
-    { title: 2, stepHeader: "Marker title 2" },
-    { title: 3, stepHeader: "Marker title 3" }
-  ];
+  steps: string[] = [];
 
-  @property({ type: Number })
-  activeStep: number;
+  /** The current state of active step. Defaults to 0 */
+  @property({ type: Number, reflect: true })
+  activeStep = 0;
 
-  connectedCallback() {
-    super.connectedCallback();
-  }
+  /**Gets or sets the default activeStep used to reset this element. The initial value corresponds to the one originally specified in the HTML that created this element. */
+  @defaultValue("activeStep")
+  defaultActiveStep = 0;
 
-  public incrementStep() {
+  /** Moves the active step forward one step */
+  public nextStep() {
+    this.emit("sgds-next-step");
     if (this.activeStep < this.steps.length - 1) {
       this.activeStep++;
     }
   }
 
-  public decrementStep() {
+  /** Moves the active step back one step */
+  public previousStep() {
+    this.emit("sgds-previous-step");
+
     if (this.activeStep > 0) {
       this.activeStep--;
     }
   }
 
+  /** Changes the active step to the last step */
   public lastStep() {
+    this.emit("sgds-last-step");
     if (this.activeStep !== this.steps.length - 1) {
       this.activeStep = this.steps.length - 1;
     }
   }
 
+  /** Changes active step to the first step */
   public firstStep() {
+    this.emit("sgds-first-step");
     if (this.activeStep > 0) {
       this.activeStep = 0;
     }
   }
 
+  /** Resets the Stepper to its initial active step state */
+  public reset() {
+    this.emit("sgds-reset");
+    this.activeStep = this.defaultActiveStep;
+  }
+
+  /**@internal */
   _onStepperItemClick(index: number) {
+    //emit an event before moving to next step
     if (this.activeStep > index) {
       this.activeStep = index;
     }
   }
+
+  /**@internal */
+  @watch("activeStep", { waitUntilFirstUpdate: true })
+  _handleActiveStepChange() {
+    this.emit("sgds-arrived");
+  }
+  
+  /**@internal */
+  _handleKeyDown(event: KeyboardEvent, index: number) {
```

Comment:
Other than enter key, do we need arrow keys (left, right) to select thru the markers instead of just relying on buttons?

Reply:
use tab to navigate the steppers is good enough for now. Left right arrow feature can be put to a enhancement ticket to do next time